server/server.go

```diff
@@ -53,61 +65,52 @@ func New(config Config) (microserver.Server, error) {
 		}
 	}
 
-	newServer := &server{
+	s := &Server{
 		// Dependencies.
-		logger: config.MicroServerConfig.Logger,
+		logger: config.Logger,
 
 		// Internals.
-		bootOnce:     sync.Once{},
-		config:       config.MicroServerConfig,
+		bootOnce: sync.Once{},
+		config: microserver.Config{
+			Logger:      config.Logger,
+			ServiceName: config.ProjectName,
+			Viper:       config.Viper,
+
+			Endpoints: []microserver.Endpoint{
+				endpointCollection.Healthz,
+				endpointCollection.Version,
+			},
+			ErrorEncoder: errorEncoder,
+		},
 		shutdownOnce: sync.Once{},
 	}
 
-	// Apply internals to the micro server config.
-	newServer.config.Endpoints = []microserver.Endpoint{
-		endpointCollection.Healthz,
-		endpointCollection.Version,
-	}
-	newServer.config.ErrorEncoder = newServer.newErrorEncoder()
-
-	return newServer, nil
+	return s, nil
 }
 
-type server struct {
-	// Dependencies.
-	logger micrologger.Logger
-
-	// Internals.
-	bootOnce     sync.Once
-	config       microserver.Config
-	shutdownOnce sync.Once
-}
-
-func (s *server) Boot() {
+func (s *Server) Boot() {
 	s.bootOnce.Do(func() {
 		// Here goes your custom boot logic for your server/endpoint/middleware, if
 		// any.
 	})
 }
 
-func (s *server) Config() microserver.Config {
+func (s *Server) Config() microserver.Config {
 	return s.config
 }
 
-func (s *server) Shutdown() {
+func (s *Server) Shutdown() {
 	s.shutdownOnce.Do(func() {
 		// Here goes your custom shutdown logic for your server/endpoint/middleware,
 		// if any.
 	})
 }
 
-func (s *server) newErrorEncoder() kithttp.ErrorEncoder {
-	return func(ctx context.Context, err error, w http.ResponseWriter) {
-		rErr := err.(microserver.ResponseError)
-		uErr := rErr.Underlying()
+func errorEncoder(ctx context.Context, err error, w http.ResponseWriter) {
```

Comment:
This can actually be defaulted in microkit. Lets do this at some point. In the operators this is common and could just be set on construction.